### ISSUE TITLE
Sourcemaps not working

### ISSUE THREAD

**Issue by berlindave** [ahmadawais/WPGulp#130]
### Expected Behavior

Displaying the correct sources in Chrome Inspector.

### Actual Behavior

Displaying just partly main scss file (which is wrong) and partly the copiled css file as sources.

![wrong-sources](https://user-images.githubusercontent.com/16339038/47512865-5c8c2480-d87d-11e8-80ce-5772be607bf7.png)

### Steps to Reproduce

1. `npm start`

```
> your-project@1.0.0 start /Users/dave/Documents/MAMP/htdocs/vor-druck.test/wp-content/themes/vordruck
> gulp

[17:44:32] Requiring external module @babel/register
[17:44:34] Using gulpfile ~/Documents/MAMP/htdocs/vor-druck.test/wp-content/themes/vordruck/gulpfile.babel.js
[17:44:34] Starting 'default'...
[17:44:34] Starting 'styles'...
[17:44:34] Starting 'vendorsJS'...
[17:44:34] Starting 'customJS'...
[17:44:34] Starting 'images'...
[17:44:34] Starting 'browsersync'...
[17:44:34] Starting '<anonymous>'...
[17:44:34] Finished 'browsersync' after 319 ms
[17:44:34] Finished 'styles' after 374 ms
[Browsersync] Proxying: http://vor-druck.test
[Browsersync] Access URLs:
 ---------------------------------------
       Local: http://localhost:3000
    External: http://192.168.178.20:3000
 ---------------------------------------
          UI: http://localhost:3001
 UI External: http://localhost:3001
 ---------------------------------------
[17:44:36] gulp-notify: [Gulp notification]

✅  ===> CUSTOM JS — completed!

[17:44:36] Finished 'customJS' after 2.46 s
[17:44:38] gulp-notify: [Gulp notification]

✅  ===> VENDOR JS — completed!

[17:44:38] Finished 'vendorsJS' after 4.41 s
[17:44:41] gulp-imagemin: Minified 0 images
[17:44:42] gulp-notify: [Gulp notification]

✅  ===> IMAGES — completed!

[17:44:42] Finished 'images' after 7.98 s
```

1. `node -v`: v10.12.0
2. `npm -v`: 6.4.1
3. WPGulp V. 2.8.0

Then, specify:

1. Operating system: OSX 10.11.6
2. Browser and version (if relevant): Google Chrome (Mac) Version 69.0.3497.100

Any suggestions what's going wrong?

Besides this little flaw I'm really glad working with WPGulp!

**Comment by Alecto** [comment]
similar problem. The map is not built correctly.

**Comment by parys-github** [comment]
I noticed that I have exactly the same problem. SourceMap only recognizes the main style file. It does not recognize the structure of individual .scss files.
My specs.
node -v: v10.12.0
npm -v: 6.4.1
WPGulp V. 2.8.0
Operating system: OSX 10.13.6

**Comment by shadtek** [comment]
> I did some further testing. I guess the reason is a problem with gulp-autoprefixer ([as described here](https://github.com/sindresorhus/gulp-autoprefixer/issues/108)). Removing the piping to autoprefixer solves the problem (except having a new one without autoprefixer).
> 
> ```
> // .pipe( autoprefixer( config.BROWSERS_LIST ) )
> ```

@berlindave Are you trying to say that this is a temp fix? I tried commenting out the line like you suggested but that didn't do anything.

**Comment by commwork** [comment]
I had this problem a while ago, too. Some mixin was the problem. Maybe check that.

**Comment by berlindave** [comment]
> @berlindave Are you trying to say that this is a temp fix?

Exactly.

> @berlindave I tried commenting out the line like you suggested but that didn't do anything.

I'm sorry to read that. Very strange. That's what I got after commenting out that line
 as described (compare to screenshot in the initial post):

![screenshot](https://user-images.githubusercontent.com/16339038/49002509-110cb500-f160-11e8-9031-7aee55923692.png)

**Comment by commwork** [comment]
Tried the new NPX Gulp now, too. Same problem. Doing this:
`// .pipe( autoprefixer( config.BROWSERS_LIST ) )`
Does not help.

Node: 9.11.2
NPM: 6.4.1
Newsest Chrome + FF
Windows 10

**Comment by ahmadawais** [comment]
Happy to accept a fix via PR. I am on vacations.

>

**Comment by zuopengz** [comment]
More clues:
At the bottom of the generated css files, there are 2 sourceMappingURL.
After removing the 2nd one manually, it works fine.  
/*# sourceMappingURL=data:application/json;charset=utf8;base64,eyJ2ZXJzaW9uIjoz ...
/*# sourceMappingURL=style.css.map */

**Comment by parys-github** [comment]
[berlindave](https://github.com/berlindave), your solution working good but I suppose this is an only temporary solution because you losing all browsers prefixes.  Honestly not really useful :-/
Did anyone try to figure out how to fix it?

**Comment by antonkeller** [comment]
I simply commented line 128 & 129 and things just worked :)
//.pipe( sourcemaps.write({ includeContent: false }) )
//.pipe( sourcemaps.init({ loadMaps: true }) )

Anything I should know with this quick fix?

**Comment by mware** [comment]
@antonkeller fix worked for me on at least one test project. 
Thx

**Comment by berlindave** [comment]
@antonkeller Great, worked also for me in one test project so far!

**Comment by berlindave** [comment]
> [berlindave](https://github.com/berlindave), your solution working good but I suppose this is an only temporary solution because you losing all browsers prefixes. Honestly not really useful :-/
Yes, of course, it's just whatt I call a "cheesy workaround", far away from a real solution.
Did you try the solution by @antonkeller? https://github.com/ahmadawais/WPGulp/issues/130#issuecomment-460978784

**Comment by keramzyt** [comment]
Same here but solution by @antonkeller is working fine and not loosing autoprefixer functionality.
Strange is that code even without fix works fine on Safari dev tools on OSX 10.14.3

**Comment by mkreckovic** [comment]
I had the same problem, and I solved it by adding `.pipe(sourcemaps.identityMap())` after `.pipe(sourcemaps.init())`.

**Comment by jmachub** [comment]
Works fine with firefox, but not with chrome. Tried all of the solutions above to no avail.

**Comment by Elschnuppero** [comment]
Doesnt work on FF nor Chrome.

**Comment by DeoThemes** [comment]
Same here, none of the solutions worked for me. When I run npm start it compiles everything, but my browser doesn't see any of the source maps, it's simply pointing to styles.min.css. I think the problem here is that source map is created for regular unminified CSS, while  there is source map for minified CSS.

**Comment by sstruemph** [comment]
I am interested in helping resolve this bug. I use WPGulp on a daily basis and have tried following the suggestions here and the only fix basically breaks Autoprefixer. Unless I am not understanding and there's a workaround for Autoprifixer?

I tried re-writing the CSS task to troubleshoot and it quickly became a rabbit hole. Several packages needed to be added and swapped out. A new autoprefixer was needed. Etc. But I gave up because even then I couldn't get it to work.

Is there anything in particular that is needed to move this forward? I would love to have sourcemaps 💯

**Comment by grayayer** [comment]
As AntonKeller wrote
> I simply commented line 128 & 129 and things just worked :)
> //.pipe( sourcemaps.write({ includeContent: false }) )
> //.pipe( sourcemaps.init({ loadMaps: true }) )

By passing false to includeContent, you're telling the sourcemaps to use the original files.

According to the gulp-sourcemaps documentation  "Including the content is the recommended way, because it "just works". When setting this to false you have to host the source files and set the correct sourceRoot."

Since wpgulp isn't including a sourceRoot value, then of course this isn't going to work.

commenting out .pipe( sourcemaps.init({ loadMaps: true }) ) is also the right thing to do because we don't want to load existing source maps. We want to load the sourcemaps that we just created.

**Comment by alirexa** [comment]
This bug still exists and fix from @antonkeller worked for me as well! 
@ahmadawais I will make a PR for this soon!

**Comment by Harry-Cowshed** [comment]
Any one find an alt fix? @tony-keller fix didn't work for me.

**Comment by datalooper** [comment]
This maybe a bonehead move, but during development, I changed my functions.php to enqueue the pre-minified style.css and it worked. 
		wp_enqueue_style( 'styles', MY_THEME_URL . '/style.css', array(), MY_THEME_VER, 'all' );
I'll have to just change it back to minified for production.
Is there any reason the gulp file specifies the minification step after the sourcemap? It seems to mess things up.
